cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -3672,33 +3672,28 @@
   </property>
 
   <!-- Retry configuration for task worker -->
+  <!-- 0 delay helps when there are free worker pods available -->
   <property>
     <name>task.worker.retry.policy.base.delay.ms</name>
-    <value>500</value>
+    <value>0</value>
     <description>
       The base delay between retries in milliseconds
     </description>
   </property>
 
-  <property>
-    <name>task.worker.retry.policy.max.delay.ms</name>
-    <value>3000</value>
-    <description>
-      The maximum delay between retries in milliseconds
-    </description>
-  </property>
-
+  <!-- Set to Integer max value, so the retry attempts end based on task.worker.retry.policy.max.time.secs -->
   <property>
     <name>task.worker.retry.policy.max.retries</name>
-    <value>10</value>
+    <value>2147483647</value>
```

Comment:
This retry setting looks scary, could you please explain better?

Reply:
There are three levels of retry checks - first a numeric limit(no of retries), then a time limit and then a strategy(fixed/exponential backoff). Here the numeric limit is set to max , so that we can base the retry on time limit. https://github.com/cdapio/cdap/blob/da58037c74af26ef26d832435b328782bbc3c443/cdap-common/src/main/java/io/cdap/cdap/common/service/RetryStrategies.java#L154

Reply:
I mean settings as a whole. Any retry with 0 delay and unlimited retries is suspicious

Reply:
Also if you plan to retry on time only, do you need this setting at all? Should it be just fixed from the code?

Reply:
As discussed offline set the retry delay to 1.